src/xma/src/xmaplugin/xmaplugin.cpp

```diff
@@ -308,7 +229,7 @@ int32_t xma_check_device_buffer(XmaBufferObj *b_obj) {
         xma_logmsg(XMA_ERROR_LOG, XMAPLUGIN_MOD, "xma_check_device_buffer failed. XMABufferObj failed allocation");
         return XMA_ERROR;
     }
-    if (b_obj_priv->dev_index < 0 || b_obj_priv->bank_index < 0 || b_obj_priv->size <= 0) {
+    if (b_obj->dev_index < 0 || xrt_core::bo::group_id(b_obj_priv->xrt_bo) < 0 || b_obj_priv->xrt_bo.size() <= 0) {
```

Comment:
This is déjà vu, didn't we see this same code in xmabuffer.cpp?

Reply:
moved this code to utils area.